sympy/polys/matrices/ddm.py

```diff
@@ -958,6 +958,41 @@ def lu(a):
 
         return L, U, swaps
 
+    def qr(self):
+        """
+        Division-based QR decomposition for DDM.
+
+        Returns:
+            - Q: Orthogonal matrix as a DDM.
+            - R: Upper triangular matrix as a DDM.
+        """
+        rows, cols = self.shape
+        Q = self.copy()
+        R = self.zeros((min(rows, cols), cols), self.domain)
+
+        # Check that the domain is a field
+        if not self.domain.is_Field:
+            raise DMDomainError("QR decomposition requires a field (e.g. QQ).")
```

Comment:
Since `self.domain` is accessed a lot I would add `K = self.domain` at the top of the function.